game-core/src/main/java/games/strategy/triplea/attachments/TriggerAttachment.java

```diff
@@ -1569,35 +1608,25 @@ public static void triggerTerritoryPropertyChange(final Set<TriggerAttachment> s
               // water territories may not have an attachment, so this could be null
               throw new IllegalStateException("Triggers: No territory attachment for:" + territory.getName());
             }
-            if (newValue.equals(attachment.getRawPropertyString(property.getFirst()))) {
-              continue;
-            }
-            if (clearFirst && newValue.length() < 1) {
-              change.add(ChangeFactory.attachmentPropertyReset(attachment, property.getFirst()));
-            } else {
-              change.add(
-                  ChangeFactory.attachmentPropertyChange(attachment, newValue, property.getFirst(), clearFirst));
-            }
-            bridge.getHistoryWriter()
-                .startEvent(MyFormatter.attachmentNameToText(t.getName()) + ": Setting " + property.getFirst()
-                    + (newValue.length() > 0 ? " to " + newValue : " cleared ") + " for "
-                    + t.getTerritoryAttachmentName().getSecond() + " attached to " + territory.getName());
+
+            getPropertyChangeHistoryStartEvent(
+                t, attachment, property.getFirst(), clearFirstNewValue,
+                t.getTerritoryAttachmentName().getSecond(), territory)
+                    .ifPresent(propertyChangeEvent -> {
+                      change.add(propertyChangeEvent.getFirst());
+                      bridge.getHistoryWriter().startEvent(propertyChangeEvent.getSecond());
+                    });
```

Comment:
I've seen the `ifPresent` part twice in this class. Would extracting this lambda into a dedicated method make sense in your opinion?

Reply:
I thought about that and tried it out before, but it seems less nice due to the mutation/side-effects of the arguments. It seems nicer AFAICT if the mutation/side-effects are "higher up the call stack", with somewhat "pure" methods being used further down. For instance reg. less need for mocks and the like for those "pure" methods. That said, it is (currently at least) so trivial that it might well make sense to extract it.
I will extract it if you think it is a good idea.

Reply:
What about just extracting the consumer part, so that you have a private helper method that returns a consumer and accepts a bridge and a change object?

Reply:
I would still be a little bit iffy, mostly due to me liking FP (functional programming), though it is a minor thing and can always be changed later. Besides, it would be a very small method without any logic, and the other method, `getPropertyChangeHistoryStartEvent`, is still free of side-effects and is easy to reason about as well as unit test. I have changed it to your suggestion.